js_modules/dagit/packages/core/src/assets/AssetEventMetadataEntriesTable.tsx

```diff
@@ -0,0 +1,92 @@
+import styled from 'styled-components/macro';
+
+import {MetadataEntry} from '../metadata/MetadataEntry';
+import {titleForRun} from '../runs/RunUtils';
+
+import {AssetMaterializationFragment} from './types/AssetMaterializationFragment';
+import {AssetObservationFragment} from './types/AssetObservationFragment';
+
+/**
+ * This component shows the metadata entries attached to an Asset Materialization or Observation event.
+ * AssetNodes also have definition-time metadata, which is unrelated to this event metadata.
+ */
+export const AssetEventMetadataEntriesTable: React.FC<{
+  event: AssetObservationFragment | AssetMaterializationFragment | null;
+  observations?: (AssetObservationFragment | AssetMaterializationFragment)[];
+}> = ({event, observations}) => {
+  if (!event || (!event.metadataEntries.length && !observations?.length)) {
+    return <Caption color={Colors.Gray500}>No materializations</Caption>;
+  }
+
+  const {metadataEntries, timestamp} = event;
+
+  return (
+    <DetailsTable>
+      <tbody>
+        {metadataEntries.map((entry) => (
+          <tr key={`metadata-${entry.label}`}>
+            <td>
+              <Mono>{entry.label}</Mono>
+            </td>
+            <td>
+              <Mono>
+                <MetadataEntry entry={entry} expandSmallValues={true} />
+              </Mono>
+            </td>
+            <td style={{opacity: 0.7}}>{entry.description}</td>
+          </tr>
+        ))}
+
+        {(observations || []).map((obs) => (
+          <React.Fragment key={obs.timestamp}>
+            {obs.metadataEntries.map((entry) => (
+              <tr key={`metadata-${obs.timestamp}-${entry.label}`}>
+                <td>{entry.label}</td>
+                <td>
+                  <MetadataEntry entry={entry} expandSmallValues={true} />
+                </td>
+                <td style={{opacity: 0.7}}>
+                  <Box flex={{gap: 8, alignItems: 'center'}}>
+                    <Icon name="observation" size={16} />
+                    <span>
+                      {`${obs.stepKey} in `}
+                      <Link to={`/instance/runs/${obs.runId}?timestamp=${obs.timestamp}`}>
+                        <Mono>{titleForRun({runId: obs.runId})}</Mono>
+                      </Link>
+                    </span>
+                  </Box>
+                  <Caption style={{marginLeft: 24}}>
+                    {` (${moment(Number(obs.timestamp)).from(Number(timestamp), true)} later)`}
+                  </Caption>
+                  {entry.description}
+                </td>
+              </tr>
+            ))}
+          </React.Fragment>
+        ))}
+      </tbody>
+    </DetailsTable>
+  );
+};
+
+const DetailsTable = styled.table`
+  width: 100%;
+  border-spacing: 0;
+  border-collapse: collapse;
+  tr td:first-child {
+    max-width: 300px;
+    width: 25%;
+  }
+  tr td {
+    border: 1px solid ${Colors.KeylineGray};
+    padding: 8px 12px;
+    font-size: 14px;
+  }
+`;
```

Comment:
- Can this table be full-bleed?
- The vertically-centered contents of each cell look a little off to me, can those be top-aligned with some top padding?

Reply:
Ahh I totally agree with the vertical centering, will fix!

I tried out the full bleed effect (Screenshot below) and I'm a little worried it "lifts" the table out of list + detail layout a bit. Going to leave that for now and we can see what Josh thinks! Just want to make sure people know it's inside the right panel and not a separate panel. 

![image](https://user-images.githubusercontent.com/1037212/197923671-72f9f894-7b15-4fd8-810b-d67030c366d7.png)